test/cases/compile_errors/compile_time_undef_ptr_cast.zig

```diff
@@ -0,0 +1,13 @@
+export fn foo() void {
+    comptime {
+        var opt_ptr: *u32 = undefined;
+        const ptr = @ptrCast(*i32, opt_ptr);
+        _ = ptr;
+    }
+}
+
+// error
+// backend=llvm
+// target=native
+//
+// :4:36: error: use of undefined value here causes undefined behavior
```

Comment:
This should not be an error.

Reply:
The stage1 compiler currently issues an error for this. Is that not according to spec?

Reply:
Stage1 is wrong, see #1947:
> Expressions which have no side effects and no possible undefined behavior, and one or more of the operands has an undefined value which is read, the expression result is an undefined value.

Reply:
According to Andrew, this should actually cause a compile error, due to the fact that a value of `undefined` indicates any value including 0, which, ptrCast should panic if that occurs.